File: Exercice 6/main.ts
// Exercice 6 : Créez une classe ClientProfessionnel qui étend la classe Client et qui a comme attribut un numéro SIRET. 
// Créez une instance pour chaque classe et affichez en console l’ensemble des attributs de ces instances. 

class Client {
    public name: string;
    public accountNumber: number;

    constructor(name: string, accountNumber: number) {
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

class ClientPro extends Client {
    public siret: number;

    constructor(name: string, accountNumber: number, siret: number) {
        super(name, accountNumber);
        this.siret = siret;
    }
}

let client1: Client = new Client("Bernard Tapis", 123456);
let client2: ClientPro = new ClientPro("Alain Juppé", 524856, 16172648);
console.log(client1.name + " " + client1.accountNumber);
console.log(client2.name + " " + client2.accountNumber + " " + client2.siret);

File: Exercice 6/main.java
class Client {
    String name;
    int accountNumber;

    Client(String name, int accountNumber) {
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

class ClientPro extends Client {
    int siret;

    ClientPro(String name, int accountNumber, int siret) {
        super(name, accountNumber);
        this.siret = siret;
    }
}

class Playground {
    public static void main(String[ ] args) {
        Client bernard = new Client("Bernard Tapis", 123456);
        ClientPro alain = new ClientPro("Alain Juppé", 524856, 16172648);
        System.out.println(bernard.name + bernard.accountNumber);
        System.out.println(alain.name + alain.accountNumber + alain.siret);
    }
}